src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -83,16 +84,16 @@ protected function setDefinition($id, Definition $definition)
         }
     }
 
-    private function findClasses($namespace, $resource)
+    private function findClasses($namespace, $pattern)
     {
         $parameterBag = $this->container->getParameterBag();
-        $resource = $parameterBag->unescapeValue($parameterBag->resolveValue($resource));
+        $pattern = $parameterBag->unescapeValue($parameterBag->resolveValue($pattern));
         $classes = array();
         $extRegexp = defined('HHVM_VERSION') ? '/\\.(?:php|hh)$/' : '/\\.php$/';
         $prefixLen = null;
-        foreach ($this->glob($resource, true, $prefix) as $path => $info) {
+        foreach ($this->glob($pattern, true, $resource) as $path => $info) {
             if (null === $prefixLen) {
-                $prefixLen = strlen($prefix);
+                $prefixLen = strlen($resource->getPrefix());
```

Comment:
which is not possible inside the foreach (the collection is returned only when the iterator is empty).